lib/base/threadpool.cpp

```diff
@@ -21,10 +20,15 @@ void ThreadPool::Start()
 	boost::unique_lock<decltype(m_Mutex)> lock (m_Mutex);
 
 	if (!m_Pool) {
-		m_Pool = decltype(m_Pool)(new boost::asio::thread_pool(m_Threads));
+		StartUnlocked();
 	}
 }
 
+void ThreadPool::StartUnlocked()
+{
+	m_Pool = decltype(m_Pool)(new boost::asio::thread_pool(Configuration::Concurrency * 2u));
+}
```

Comment:
This function differs from `Start()` not only in locking and in general, it can also be given a more precise name like `InitializePool()`.